### ISSUE TITLE
Button toggling checkbox is always left un-checked on spacebar "click"

### ISSUE THREAD

**Issue by bggardner** [twbs/bootstrap#27600]
- Operating system and version: Ubuntu 18.10
- Browser and version: Chrome 70.0.3538.77 and Firefox 63.0
- Example: https://jsbin.com/gaxuwobiku/edit?html,output

Problem:
When I'm filling out forms using a physical keyboard, I typically tab from input to input, and use the spacebar for checking/unchecking checkboxes.  Unfortunately, using the [example code for a label button with "embedded" checkbox](https://getbootstrap.com/docs/4.0/components/buttons/#checkbox-and-radio-buttons) does not work when the spacebar initiates the "click".  A mouse click works as expected.  It appears to do something with the timing of the event, as the "checked" property is properly set immediately after the event, but is always removed (read as false) sometime after.

**Comment by bggardner** [comment]
It appears the checkbox gets set during the "keyup" event, which happens before the "click" event is fired, so here's a workaround:
```
$(document).on("keyup", '[data-toggle="buttons"] input[type="checkbox"]', function(e) {
  if (e.key == " ") {
    e.preventDefault(); // Prevent the spacebar from toggling and firing the click event
    $(this).closest('[data-toggle="buttons"]').button("toggle"); // Let Bootstrap toggle
  }
});
```
Not sure how to modify `button.js` for a proper fix.

**Comment by AceJJ** [comment]
For button.js, I think we just need a new keyup event type:
`const Event = {
  CLICK_DATA_API      : `click${EVENT_KEY}${DATA_API_KEY}`,
  KEYUP_DATA_API      : `keyup${EVENT_KEY}${DATA_API_KEY}`,
  FOCUS_BLUR_DATA_API : `focus${EVENT_KEY}${DATA_API_KEY} ` +
                          `blur${EVENT_KEY}${DATA_API_KEY}`
}`

Then, we could add it to the document event handler, the same as the on click event:

`/**
 * ------------------------------------------------------------------------
 * Data Api implementation
 * ------------------------------------------------------------------------
 */

$(document)
  .on(Event.CLICK_DATA_API, Selector.DATA_TOGGLE_CARROT, (event) => {
    event.preventDefault()

    let button = event.target

    if (!$(button).hasClass(ClassName.BUTTON)) {
      button = $(button).closest(Selector.BUTTON)
    }

    Button._jQueryInterface.call($(button), 'toggle')
  })
  .on(Event.KEYUP_DATA_API, Selector.DATA_TOGGLE_CARROT, (event) => {
    if (e.key == " ") {
      event.preventDefault()

      let button = event.target

      if (!$(button).hasClass(ClassName.BUTTON)) {
        button = $(button).closest(Selector.BUTTON)
      }

      Button._jQueryInterface.call($(button), 'toggle')
    }
  })
  .on(Event.FOCUS_BLUR_DATA_API, Selector.DATA_TOGGLE_CARROT, (event) => {
    const button = $(event.target).closest(Selector.BUTTON)[0]
    $(button).toggleClass(ClassName.FOCUS, /^focus(in)?$/.test(event.type))
  })`

**Comment by patrickhlauke** [comment]
late reply, but...should be fixed now by https://github.com/twbs/bootstrap/pull/28834
